scripts/junit2issue.py

```diff
@@ -0,0 +1,189 @@
+# and add custom attributes to it.
+class ZephyrTestCase(TestCase):
+    '''
+        Zephyr Customer content
+    '''
+
+def parser_testsuites_for_result(xml, test_result):
+    '''
+        parser_testsuites_for_type
+        @xml: input JUnitXml instance
+        @test_result: test type [Failure, Error, Skipped]
+        return:
+            a list of defined type cases info hash
+    '''
+    logging.info("check result with %s:", str(test_result))
+    report_list = []
+    version = "unknown"
+
+    for suite in xml:
+        # handle suites
+        print("name = {}".format(suite.name))
+        print("time = {}".format(suite.time))
+        print("tests = {}".format(suite.tests))
+        print("failures = {}".format(suite.failures))
+        print("errors = {}".format(suite.errors))
+        print("skipped = {}".format(suite.skipped))
+
+        for prop in suite.child(Properties):
+            print("{} : {}".format(prop.name, prop.value))
+            version = prop.value
+
+        for case in suite:
+            # handle cases
+            my_case = ZephyrTestCase.fromelem(case)
+            for res in my_case.result:
+                if isinstance(res, test_result):
+                    print("+")
+                    print("  " + my_case.classname)
+                    print("  " + my_case.name)
+                    print("  " + res.message)
+                    report_case = {
+                        'APP_NAME' : my_case.classname,
+                        'CASE_NAME' : my_case.name,
+                        'ZEPHYR_VERSION' : version,
+                        'RESULT' : res.message,
+                        'PLATFORM': suite.name,
+                        'LOGS': res.text,
+                    }
+                    report_list.append(report_case)
+                    # print("  " + res.text)
+    return report_list
+
+def gen_content_from_template(template, context):
+    '''
+        gen_content_from_template
+        @template: in template string
+        @context: in context
+        return:
+            generated issue content string
+    '''
+    _s = Template(template)
+    content = _s.substitute(context)
+    # logging.info(content)
+    return content
+
+def github_login():
+    '''
+        github_login
+    '''
+
+    access_token = os.getenv('GITHUB_ACCESS_TOKEN')
+
+    if not access_token:
+        print("no access token find in system environment variable\n")
+        print("please set GITHUB_ACCESS_TOKEN with:\n")
+        print("export GITHUB_ACCESS_TOKEN=<your github access token>\n")
+        return None
+
+    # using an access token
+    return Github(access_token)
+
+def check_issue_existing(repository, report):
+    '''
+        check whether the report issue exists
+        @repository: in repo target
+        @report: dict fo reporting content
+        return:
+            existing issue or None
+    '''
+    open_issues = repository.get_issues(state='open')
```

Comment:
I guess that it will take a long time because over 1000 open issues in the Zephyr project. Maybe a better way is we just search the issues, which are opened in last 6 months.

Reply:
we can not search the zephyr project for now, only your own repo, so by now it is ok. But you are right,  let me add some tine scale.